config/routes.rb

```diff
@@ -2,5 +2,10 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root 'home#index'
+  get '/users', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:user_id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
+
```

Comment:
- [ ] **[OPTIONAL]** I think we can define these routes in a more concise manner using the `resources` helper.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

Reply:
Changed as instructed